### DISCUSSION TITLE
How to download all files from an object in MinIO?

### DISCUSSION THREAD

**Discussion by pedrocasteli** [minio/minio#15798] [Q&A] [answered]
I have a bucket called "**products**" in MinIO, with different folders for different products, and inside each product folder, several XML files.
So the structure is like this: 
`products/apples/apples1.xml, apples2.xml, apples3.xml...`
and
`products/bananas/bananas1.xml, bananas2.xml, bananas3.xml...`
I need to download all product folders (apples, bananas, grapes, etc) and their XMLs.
I am using the Python SDK and the `client.fget_object()` function but it doesn't work. Does anybody know how to download a whole folder like this?

**Comment by vadmeste** [comment, marked as answer]
@pedrocasteli you need to use client.list_objects() to list object names first, then run client.fget_object() to download objects one by one. You also need to create missing folders before downloading the objects.
